app/controllers/ticket_comments_controller.rb

```diff
@@ -15,17 +15,16 @@ def index
 
   def create
     ticket = Ticket.find(params.require(:ticket_id))
-    acting_stakeholder_id = params.require(:acting_stakeholder_id)
+    acting_stakeholder = TicketStakeholder.find(params.require(:acting_stakeholder_id))
 
-    # Currently only stakeholders can create comments.
-    return head :unauthorized unless ticket.can_user_access?(current_user)
-    return head :bad_request unless ticket.user_stakeholders(current_user).map(&:id).include?(acting_stakeholder_id)
+    return head :bad_request unless ticket.user_stakeholders(current_user).map(&:id).include?(acting_stakeholder.id)
```

Comment:
In the internal code of `user_stakeholders`, you're relying on ActiveRecord `or`, which returns a "clean" association, so I wonder whether `.pluck(:id)` would work here... Please try it out and let me know 🙃